agent/hibinit-agent

```diff
@@ -118,78 +118,87 @@ def find_grub_mount():
+        echo_resume_device_cmd = "echo {swap_device} > /sys/power/resume"
+        echo_resume_device_cmd = echo_resume_device_cmd.format(swap_device=swap_device)
+        print_to_sys_log("sys/power/resume exist and would be updated")
+        check_call(echo_resume_device_cmd, shell=True)
 
     if os.path.exists("/sys/power/resume_offset"):
-       echoResumeOffsetCmd = "echo {offset} > /sys/power/resume_offset"
-       echoResumeOffsetCmd = echoResumeOffsetCmd.format(offset=offset)
-       log("sys/power/resume_offset exist and would be updated")
-       check_output(echoResumeOffsetCmd, shell=True)
-    log("GRUB configuration is updated")
+        echo_resume_offset_cmd = "echo {offset} > /sys/power/resume_offset"
+        echo_resume_offset_cmd = echo_resume_offset_cmd.format(offset=offset)
+        print_to_sys_log("sys/power/resume_offset exist and would be updated")
+        check_call(echo_resume_offset_cmd, shell=True)
+
+    print_to_sys_log("GRUB configuration is updated")
 
 
-def update_kernel_swap_offset(swapon, swapoff, filename, grub_update, btrfs_enabled):
-    swapon = swapon.format(swapfile=filename)
-    log("Running: %s" % swapon)
+@critical_process
+def update_kernel_swap_offset(config):
+    swapon = config.swapon.format(swapfile=SWAP_FILE)
+    print_to_sys_log("Running: %s" % swapon)
     check_call(swapon, shell=True)
-    log("Updating the kernel offset for the swapfile: %s" % filename)
+    print_to_sys_log("Updating the kernel offset for the swapfile: %s" % SWAP_FILE)
 
-    statbuf = os.stat(filename)
+    statbuf = os.stat(SWAP_FILE)
     dev = statbuf.st_dev
-    if btrfs_enabled:
-        getOffset = "btrfs inspect-internal map-swapfile -r {swapfile}".format(swapfile=filename)
-        offset = int(check_output(getOffset, shell=True).decode(sys.getfilesystemencoding()))
+    if config.btrfs_enabled:
+        get_offset = "btrfs inspect-internal map-swapfile -r {swapfile}".format(swapfile=SWAP_FILE)
+        offset = int(check_output(get_offset, shell=True).decode(sys.getfilesystemencoding()))
     else:
-        offset = get_file_block_number(filename)
+        offset = get_file_block_number(SWAP_FILE)
 
-    if grub_update:
-        dev_str = find_device_for_file(filename)
+    if config.grub_update:
+        dev_str = find_device_for_file(SWAP_FILE)
         patch_grub_config(dev_str, offset)
     else:
-        log("Skipping GRUB configuration update")
+        print_to_sys_log("Skipping GRUB configuration update")
 
-    log("Setting swap device to %d with offset %d" % (dev, offset))
+    print_to_sys_log("Setting swap device to %d with offset %d" % (dev, offset))
 
-    if not btrfs_enabled:
+    if not config.btrfs_enabled:
         # Set the kernel swap offset, see https://www.kernel.org/doc/Documentation/power/userland-swsusp.txt
         # From linux/suspend_ioctls.h
-        SNAPSHOT_SET_SWAP_AREA = 0x400C330D
+        snapshot_set_swap_area = 0x400C330D
```

Comment:
Moving this as a constant might be good. Comment on what  is the hex code representing ?

Reply:
The current comment seems sufficient explanation for `snapshot_set_swap_area`. 

This is a value hard-coded in the kernel itself so it is not so much what the hex code represent, but simply a value we need for our kernel swap offset to be set properly.